### PR TITLE
refactor(api): define abci error codes in protobuf

### PR DESCRIPTION
**Pull request by SuperFluffy**
## Summary
Define protobuf spec for ABCI error codes, remove sequencer-types crate.

## Background
All public APIs should be defined in `astria-api` and the protobuf spec. This patch allows non-rust code to understand the abci codes emitted by sequencer.

The abci return code convention of having `0` be success and non-zero be an error does not cleanly map to the protobuf convention. Protobuf enums should have a default value `UNSPECIFIED = 0`. We therefore go with the protobuf convention and define the enum type as `AbciErrorCode`.

## Changes
- Define ABCI errors codes in protobuf spec
- Move Rust code from api-sequencer-types to astria-api
- Delete sequencer-types crate

## Testing
Tests making use of abci codes still work.

## Related Issues
closes https://github.com/astriaorg/astria/issues/369

### CODE REVIEW

File: proto/astria/sequencer/v1alpha1/abci.proto [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+package astria.sequencer.v1alpha1;
+
+enum AbciCode {
+    OK = 0;
+    UNKNOWN_PATH = 1;
+    INVALID_PARAMETER = 2;
+    INTERNAL_ERROR = 3;
+    INVALID_NONCE = 4;
+    TRANSACTION_TOO_LARGE = 5;
```
Comment (SuperFluffy) [inline proto/astria/sequencer/v1alpha1/abci.proto:11]: @noot I renamed this from `InvalidSize` to make it slightly more descriptive.

File: proto/astria/sequencer/v1alpha1/abci.proto [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+package astria.sequencer.v1alpha1;
+
+enum AbciCode {
+    ABCI_CODE_OK = 0;
+    ABCI_CODE_UNKNOWN_PATH = 1;
+    ABCI_CODE_INVALID_PARAMETER = 2;
+    ABCI_CODE_INTERNAL_ERROR = 3;
+    ABCI_CODE_INVALID_NONCE = 4;
+    ABCI_CODE_TRANSACTION_TOO_LARGE = 5;
```
Comment (SuperFluffy) [inline proto/astria/sequencer/v1alpha1/abci.proto:11]: @noot I renamed this from `INVALID_SIZE` to make it slightly more specific/descriptive.

File: proto/astria/sequencer/v1alpha1/abci.proto [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+package astria.sequencer.v1alpha1;
+
+enum AbciErrorCode {
+  ABCI_ERROR_CODE_UNSPECIFIED = 0;
+  ABCI_ERROR_CODE_UNKNOWN_PATH = 1;
+  ABCI_ERROR_CODE_INVALID_PARAMETER = 2;
+  ABCI_ERROR_CODE_INTERNAL_ERROR = 3;
+  ABCI_ERROR_CODE_INVALID_NONCE = 4;
+  ABCI_ERROR_CODE_TRANSACTION_TOO_LARGE = 5;
```
Comment (SuperFluffy) [inline proto/astria/sequencer/v1alpha1/abci.proto:11]: @noot I renamed this from `INVALID_SIZE` to make it slightly more specific/descriptive.

File: crates/astria-conductor/src/data_availability/block_verifier.rs [hunk 4]
```diff
@@ -296,6 +296,23 @@ mod test {
 
     use super::*;
 
+    /// Constructs a `[merkle::Tree]` from an iterator yielding byte slices.
+    ///
+    /// This hashes each item before pushing it into the Merkle Tree, which
+    /// effectively causes a double hashing. The leaf hash of an item `d_i`
+    /// is then `MTH(d_i) = SHA256(0x00 || SHA256(d_i))`.
+    fn merkle_tree_from_transactions<I, B>(iter: I) -> merkle::Tree
```
Comment (SuperFluffy) [inline crates/astria-conductor/src/data_availability/block_verifier.rs:304]: no code changes here. Just moved this out of the sequencer-types crate as this was only used in conductor.

**Review by noot — APPROVED** [review]
looks good!

**Review by sambukowski — APPROVED** [review]

**Pull request merged**